### ISSUE TITLE
diff for not git-commited files

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#2999]
It feels like `diff` can do a better job on not committed changes.

```
$ dd if=/dev/random of=small.txt bs=1024000000 count=1 # modify the file
$ dvc add small.txt
$ dvc diff -t small.txt HEAD          # IT SUPPOSE TO BE DIFFERENT FROM HEAD
dvc diff from 2dae4a2 to 2dae4a2   # IT IS NOT :(  

$ dvc diff -t small.txt HEAD^
dvc diff from 8a1aa99 to 2dae4a2 .   # IT IS NOT DIFFERENT EVEN WITH OLDER VERSION:(

diff for 'small.txt'
-small.txt with md5 b4301ef665d0343e028fcd5821654edc
+small.txt with md5 b4301ef665d0343e028fcd5821654edc

file size was not changed

$ git add small.txt.dvc
$ git commit -m change
$ dvc diff -t small.txt HEAD^ . # NOW YOU CAN SEE THE DIFFERENCE
dvc diff from 2dae4a2 to 12ed32b

diff for 'small.txt'
-small.txt with md5 b4301ef665d0343e028fcd5821654edc
+small.txt with md5 4be464ee7907d9d5e558133907bb3510

added file with size 1.0 GB
```

**Comment by jorgeorpinel** [comment]
Just to clarify, this is about uncommitted changes in Git+DVC repos, right? Not about `dvc diff` working for non-Git DVC projects.

If so, it was probably addressed by #3229 

```console
# Install pre-release bc new `dvc diff` isn't released yet as of rn:
$ virtualenv -p python3 .venv
...
$ source .venv/bin/activate
$ pip install git+https://github.com/iterative/dvc
...

$ git init
$ echo ".venv" >> .git/info/exclude
$ dvc init
$ git commit -am "New DVC repo"

$ echo "data" > small.txt
$ .venv/bin/dvc diff  # DEFAULTS TO HEAD
Added:                                                                  
    small.txt

summary: added (1), deleted (0), modified (0)
dvc diff -t small.txt HEAD^  # THIS WON'T WORK HERE BECAUSE THERE'S NO HEAD^ BUT OTHERWISE IT WOULD, I THINK
ERROR: unexpected error - Cmd('git') failed due to: exit code(128)      
  cmdline: git rev-parse --short HEAD^
  stderr: 'fatal: Needed a single revision'  # THIS ERROR IS CONFUSING THOUGH, WILL OPEN SEPARATE ISSUE...


Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!

$ git add --all
$ git commit -m change
...
$ .venv/bin/dvc diff HEAD^  # ALSO WORKS, SAME AS IN ORIGINAL EXAMPLE.
Added:                                                                  
    small.txt

summary: added (1), deleted (0), modified (0)
```

**Comment by jorgeorpinel** [comment]
p.s. I'm saying I think this issue can be closed.
